deepreg/model/backbone/global_net.py

```diff
@@ -81,18 +79,16 @@ def call(self, inputs, training=None, mask=None):
         :return:
         """
         # down sample from level 0 to E
-        h_in = inputs
         for level in range(self._extract_max_level):  # level 0 to E - 1
-            h_in, _ = self._downsample_blocks[level](inputs=h_in, training=training)
+            h_in, _ = self._downsample_blocks[level](inputs=inputs, training=training)
         h_out = self._conv3d_block(
             inputs=h_in, training=training
         )  # level E of encoding
 
         # predict affine parameters theta of shape = [batch, 4, 3]
-        theta = self._dense_layer(h_out)
-        theta = self._reshape(theta)
-
+        self.theta = self._dense_layer(h_out)
+        self.theta = self._reshape(self.theta)
         # warp the reference grid with affine parameters to output a ddf
-        grid_warped = layer_util.warp_grid(self.reference_grid, theta)
+        grid_warped = layer_util.warp_grid(self.reference_grid, self.theta)
         output = grid_warped - self.reference_grid

```

Comment:
another thing, if it's not too hard to ask, can we add the shapes of the tensors after nearly each line?
e.g.
- we have a new var, then we should add the shape as comment in the end or on top
- a var has changed shape, we should add the shape as comment in the end or on top

I argue this will make the code easier to understand,
@NMontanaBrown @YipengHu what do you think ;)

Reply:
Personally, if it is not in https://ucl-candi.github.io/DeepReg/#/CONTRIBUTING
no one should force it.

Reply:
is it in PEP8? @mathpluscode

Reply:
no no, but it will be really helpful for developing deep learning codes, we can discuss it in the meeting.
manipulating tensors without knowing their shapes are hard :(

Reply:
another point is declaring the input/output types, although unlike other language it's not mandatory in python, with py3.7, 3.8 etc, they are actually adding more supports for this ;) we can also discuss this in meeting